### PR TITLE
support catalog containing artifacts

### PR DESCRIPTION
**Pull request by pjain1**
Runtime should pass catalog containing artifacts and schema to the sql lib. Format for this catalog is like -
```
{
  "artifacts": [
    {
      "name": "...",
      "type": "...",
      "payload": "..."
    },
    ...
  ],
  "schemas": [
    {
      "name": "...",
      "tables": [
        {
          "name": "...",
          "columns": [
            {
              "name": "...",
              "type": "..."
            },
            ...
          ]
        }
      ]
    }
  ]
}
```
An example given [here](https://github.com/rilldata/rill-developer/pull/1057/files#diff-8a27ac90cca68c038fbde129e3348dd7dd03b75de026f3f936dc4781017c0ced).

### CODE REVIEW

File: sql/src/main/java/com/rilldata/calcite/models/ArtifactStore.java [hunk 1]
```diff
@@ -2,31 +2,42 @@
...
+  }
+
+  public void saveArtifact(Artifact artifact)
   {
     artifacts.computeIfAbsent(artifact.getType(), type -> new HashMap<>());
     Artifact existing = artifacts.get(artifact.getType()).putIfAbsent(artifact.getName().toLowerCase(), artifact);
     if (existing != null) {
-      System.out.printf("Artifact with name %s of type %s already exists, overwriting!%n", artifact.getName().toLowerCase(),
+      System.out.printf("Artifact with name %s of type %s already exists, overwriting!\n",
+          artifact.getName().toLowerCase(),
```
Comment (begelundmuller) [inline sql/src/main/java/com/rilldata/calcite/models/ArtifactStore.java:29]: Will this lead to logs when validating a metrics view that already exists? That will happen when people want to update metrics views. Product isn't happy about unexpected logs during normal usage

**Review by begelundmuller — CHANGES_REQUESTED** [review]
Looks good, thank you for this PR.

One note: it looks like the new catalog format isn't supported in the requests/rpc entrypoint. Can we add that?

Two questions:

- How can we specify a default schema? So we don't have to do `main.` for example. Maybe add `defaultSchema` as a top-level option on the input catalog?
- Where can I get a list of type names supported for the schema?

**Review by begelundmuller — APPROVED** [review]

**Comment by begelundmuller** [comment]
Merging now, but please consider the comments above when you have time

File: sql/src/main/java/com/rilldata/calcite/models/ArtifactStore.java (hunk 1 above)
Comment (pjain1) [reply sql/src/main/java/com/rilldata/calcite/models/ArtifactStore.java:29]: removed this in `runtime-metrics` PR

**Comment by pjain1** [comment]
> How can we specify a default schema? So we don't have to do main. for example. Maybe add defaultSchema as a top-level option on the input catalog?

There are two things - first we can add a field to the catalog to specify default schema, this is simple. Second, I need to look into it if calcite supports such thing, otherwise will need to manually prefix default schema name to query if not specified by user. 

> Where can I get a list of type names supported for the schema?

It should support all duckdb types, conversion from duckdb to sql standard types is done [here](https://github.com/rilldata/rill-developer/blob/main/sql/src/main/java/com/rilldata/calcite/StaticTable.java#L30). Similar thing will need to be done for druid then schema will also need to contain database name i.e. duckdb or druid.

**Pull request merged**